### PR TITLE
Remove deprecated methods for 1.0

### PR DESCRIPTION
**Pull request by juanjoDiaz**
Deprecated methods were piling up. I know it's never nice for the user to remove them. However, I think that a 1.0 release should be a clean release without deprecated methods.

Wdyt?

### CODE REVIEW

**Comment by juanjoDiaz** [comment]
I see...
I guess I was too optimistic 😄 

Feel free to keep this so it can be merge when 1.0 is closer to a release or close it and I can reopen in the future.

**Comment by ennru** [comment]
What about removing deprecations introduced before 0.18 right now. And leave the newer ones during the 1.0 milestones?

**Comment by 2m** [comment]
Sounds good!

**Comment by juanjoDiaz** [comment]
PR updated to remove only the deprecated stuff before 0.18.
Just let me know if you want me to delete the ones from 0.20 or 1.0-M1 and I'll be happy to do so 😄 

I've noticed few more inconsistencies around deprecations:
- There is a bunch of methods in the Solr connectors marked as `@deprecated` in the javadoc comment but missing the actual `@deprecated` annotation.
- There are some methods in the MQTT connector and others that are missing the `@java.lang.Deprecated` annotations.
- There are some methods having the `@deprecated` in the javadoc and other missing it.

File: cassandra/src/main/scala/akka/stream/alpakka/cassandra/javadsl/CassandraFlow.scala [hunk 1]
```diff
@@ -24,16 +24,6 @@ object CassandraFlow {
       .createWithPassThrough[T](parallelism, statement, (t, p) => statementBinder.apply(t, p))(session)
       .asJava
 
-  @deprecated("use createWithPassThrough without ExecutionContext instead", "0.20")
```
Comment (ennru) [inline cassandra/src/main/scala/akka/stream/alpakka/cassandra/javadsl/CassandraFlow.scala]: Leave this for now, please.

File: cassandra/src/main/scala/akka/stream/alpakka/cassandra/javadsl/CassandraFlow.scala [hunk 2]
```diff
@@ -54,27 +44,4 @@ object CassandraFlow {
...
-
-  /**
-   * Creates a flow that batches using an unlogged batch. Use this when most of the elements in the stream
-   * share the same partition key. Cassandra unlogged batches that share the same partition key will only
-   * resolve to one write internally in Cassandra, boosting write performance.
-   *
-   * Be aware that this stage does not preserve the upstream order.
-   */
-  @deprecated("use createUnloggedBatchWithPassThrough without ExecutionContext instead", "0.20")
-  def createUnloggedBatchWithPassThrough[T, K](parallelism: Int,
```
Comment (ennru) [inline cassandra/src/main/scala/akka/stream/alpakka/cassandra/javadsl/CassandraFlow.scala]: Leave this, please.

File: jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala [hunk 3]
```diff
@@ -5,20 +5,29 @@
 package akka.stream.alpakka.jms.scaladsl
 
 import java.nio.charset.Charset
+<<<<<<< Updated upstream
```
Comment (ennru) [inline jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala]: Unfinished merge?

**Comment by ennru** [comment]
Thank you for this housekeeping.

**Pull request merged**